### PR TITLE
Add ahmed story

### PR DESCRIPTION
**Pull request by mooreds**

### CODE REVIEW

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+<!--more-->
+
+*This interview has been lightly edited for clarity and length.*
+
+-------
+
+**Dan:** Tell me a bit about your work as a developer and what OGGEH Cloud Computing is working on?
+
+**Ahmed:** I consider myself a front end Swiss Army Knife. I started as a UX Designer 16 years ago. My developer journey started with ActionScript producing rich internet apps in Flash Player. The turning point after the famous letter by Steve Jobs, which was officially the death certificate of the Flash Player. Right then, I switched to JavaScript.
```
Comment (dpolstra) [inline site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:22]: The turning point **_came_** after

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+OGGEH Cloud Computing the only qualified Progressive Web Application (PWA) Agency by Google Developers in Africa and the MENA region. As well as the only Google Cloud Technology Partner in the Arabian countries! Actual users/customers are always looking for a simple way to manage their content at the backend. Something that does not involve writing mysterious markup like HTML and/or weird shortcodes as most plugins do (for WordPress, Joomla, Drupal, and others).
```
Comment (dpolstra) [inline site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:24]: OGGEH Cloud Computing **_is_** the only qualified

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+OGGEH Cloud Platform takes care of complex backend/infrastructure logic for security and scalability. 
+
+This allows frontend designers/developers to build their apps without requiring them to build an API and/or server-side services. Additionally, the platform provides a ready-made Cloud CMS that does pure content management. Not a website builder and/or a design tool.
+
+We pick our early adopters carefully. Large organizations with real traffic on PWA(s). Many members use OGGEH Cloud Platform, such as [Omar Effendi stores](https://shop.omareffendi.com.eg/?lang=en), one of the oldest and largest retailers in Egypt (founded in 1856).
+
+I would say that OGGEH Cloud Computing is a business built by nerds seeking out other nerds. Surviving on cloud usage subscriptions. (We might expose our platform as an open-source project in the near future.)
+
+> [FusionAuth] works like a charm from the first attempt. That's something that doesn't usually happen when trying new things.
```
Comment (dpolstra) [inline site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:34]: FusionAuth] work**_ed_** ~s~ like a charm

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+**Ahmed:** Yes, that's exactly what it does. There's a [single-page](https://docs.oggeh.com) that covers our API reference, developer access, service architecture, and more. We have [a short video](https://dev.oggeh.com/) that presents OGGEH Cloud Platform for the developer community as well.
+
+**Dan:** How do you use FusionAuth? OAuth? User management? Social sign-on? Something else?
+
+**Ahmed:** As we speak, FusionAuth manages and authenticates our users to use OGGEH Cloud Platform tools (Cloud Apps, Cloud CMS, Cloud Stats and others). As well as a separate tenant/customer for each cloud app that requires its user authentication at the frontend (like e-commerce apps).
+
+We prefer OpenID Connect and OAuth 2.0 for the implementation. At the moment, we use both Authorization Code and Implicit flows for the authentication of our platform tools, and only Implicit flow on cloud apps (since they're progressive web apps).
+
+Securing OGGEH Cloud Platform customers are one of our main focuses, so we do not allow any special markup and/or scripts anyway. As I mentioned earlier, we introduce pure content management only. This means there's no easy way to inject code using our platform.
```
Comment (dpolstra) [inline site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:46]: Platform customers ~are~ **_is_** one

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+**Dan:** Are you creating the tenants for each customer via the API or in the UX? Is the main reason for that separate because of the look and feel differences or are there other reasons to have a tenant for each cloud app?
```
Comment (dpolstra) [inline site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:48]: Is the main reason for that separate (separation?)

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+There's also one important note I must add here. Gluu might break due to expired JKS files for SCIM. These Java Keystore files are generated upon installation and [expire after one year](https://gluu.org/docs/gluu-server/4.0/operation/replace-expired-jks-scim/). Imagine how we felt after exactly one year of the installation to find a broken Gluu and spending many hours digging up this annoying fact!
+
+To be honest, that was the moment where I decided to get rid of Gluu. There were many other alternatives. FusionAuth caught my attention and just stuck there. I used to say that Auth0 has the best documentation and comprehensive service. But all that changed dramatically after I explored FusionAuth. It works like a charm from the first attempt. That's something that doesn't usually happen when trying new things. I wish to thank you for that.
```
Comment (dpolstra) [inline site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:70]: it worked like a charm

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+We need to have a stateless container. I'm not sure if I can auto-install a custom plugin using a kickstart file by FusionAuth. [ed note: This is not currently possible, but there's an [open issue](https://github.com/FusionAuth/fusionauth-issues/issues/1096).]
+
+**Dan:** Any general feedback/areas to improve?
+
+**Ahmed:** Would love to be able to change the branding of the default login screen of FusionAuth from the very first setup/kickstart. Even though we use a custom tenant for our users. Separating those from FusionAuth admin interface (which works great). Someone might just hit the hostname of our authentication server directly without any request parameters that define a tenant (i.e. `sso.oggeh.com`). 
+
+They will not be able to login using their credentials of course, and I do not mind that the FusionAuth logo sits there at the bottom and the favicon. But that login screen has this "help" link at the top-right corner which links to the FusionAuth docs and I prefer to show the OGGEH logo instead. It would be nice if we can use a custom stylesheet as well.
+
```
Comment (dpolstra) [inline site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:89]: maybe add a note about this being available in the paid editions?

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md (hunk 5 above)
Comment (jobannon) [inline site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:48]: separation

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+**Dan:** What problems did we solve for you? And how were you solving them before FusionAuth?
+
+**Ahmed:** Two words: customization, and scalability! FusionAuth provides a very comforting way to organize tenants, applications, identify providers, email templates, and themes. In addition to the fact that the footprint is remarkably small and does not require great CPU or memory resources to operate. Which fits perfectly into our scalable stack.
+
+Earlier, we had to replicate the authentication server for each customer (Cloud App) just to customize the login theme and email templates. We also had to use separate servers on each due to the high resource requirements to start and operate. Scaling servers is far more complex and comes with a greater financial cost.
+
+> Literarily 40% of our expenses [are] just gone [after adopting FusionAuth].
```
Comment (jobannon) [inline site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:58]: 🎉  from the ⛰️ 's

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+**Dan:** How much time and money would you say FusionAuth has saved you?
+
+**Ahmed:** A tremendous amount! Literarily 40% of our expenses just gone. No more unnecessary high resources for Gluu. FusionAuth works just fine on our auto-scaled stack.
+
+> [FusionAuth's] footprint is remarkably small and does not require great CPU or memory resources to operate.
+
+**Dan:** How do you run FusionAuth (k8s, standalone tomcat server, behind a proxy, etc)?
+
+**Ahmed:** K8s. We do the same for the rest of OGGEH Cloud Platform tools, which includes a variety of auto-scaled micro-services. Now FusionAuth is one of those. I did not exactly pick the docker version provided by FusionAuth docs. I had to modify the official Dockerfile from FusionAuth/fusionauth-containers repository for auto-installing a custom plugin. One that handles LDAP Salted SHA512 passwords (imported from Gluu).
```
Comment (jobannon) [inline site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:80]: wonder if this is a bit of compromising information or if it is inconsequential `LDAP Salted SHA512 passwords`

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+**Ahmed:** K8s. We do the same for the rest of OGGEH Cloud Platform tools, which includes a variety of auto-scaled micro-services. Now FusionAuth is one of those. I did not exactly pick the docker version provided by FusionAuth docs. I had to modify the official Dockerfile from FusionAuth/fusionauth-containers repository for auto-installing a custom plugin. One that handles LDAP Salted SHA512 passwords (imported from Gluu). 
+
+We need to have a stateless container. I'm not sure if I can auto-install a custom plugin using a kickstart file by FusionAuth. [ed note: This is not currently possible, but there's an [open issue](https://github.com/FusionAuth/fusionauth-issues/issues/1096).]
+
+**Dan:** Any general feedback/areas to improve?
+
+**Ahmed:** Would love to be able to change the branding of the default login screen of FusionAuth from the very first setup/kickstart. Even though we use a custom tenant for our users. Separating those from FusionAuth admin interface (which works great). Someone might just hit the hostname of our authentication server directly without any request parameters that define a tenant (i.e. `sso.oggeh.com`).
```
Comment (jobannon) [inline site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:86]: Not sure if I completely follow this.

**Review by jobannon — COMMENTED** [review]
Interesting read!

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md (hunk 8 above)
Comment (jobannon) [reply site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:58]: In fact, I think this should be a banner stat!

Comment (mooreds) [reply site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:58]: Awesome point, I'll change the description. We do lead with this on the customers page.

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md (hunk 9 above)
Comment (mooreds) [reply site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:80]: Hmmm. He didn't seemed concerned about sharing it. Maybe he is rehashing the passwords?

File: site/_posts/2021-04-13-oggeh-fusionauth-gluu.md (hunk 10 above)
Comment (mooreds) [reply site/_posts/2021-04-13-oggeh-fusionauth-gluu.md:86]: See https://github.com/FusionAuth/fusionauth-issues/issues/994

Comment (mooreds) [reply site/_posts/2021-04-13-oggeh-fusionauth-gluu.md]: I also tweaked the wording a bit.

**Pull request merged**